packages/contracts-bedrock/contracts/echidna/FuzzBurn.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity 0.8.15;
+
+import { Burn } from "../libraries/Burn.sol";
+
+contract EchidnaFuzzBurn {
+    bool failedEthBurn;
+    bool failedGasBurn;
+
+    /**
+     * @notice Takes an integer amount of eth to burn through the Burn library and
+     * updates the contract state if an incorrect amount of eth moved from the contract
+     */
+    function testBurn(uint256 _value) public {
+        // cache the contract's eth balance
+        uint256 preBurnBalance = address(this).balance;
+
+        // execute a burn of _value eth
+        // (may way to add guardrails to this value rather than a truly unbounded uint256)
+        Burn.eth(_value);
+
+        // check that exactly _value eth was transfered from the contract
```

Comment:
Do reverts for underflow not matter here?

Reply:
Oh, that's a great question. I think it's probably unreachable here, but regardless we should account for it.

Reply:
To address this I: 
1. Added a `bound()` to prevent overflow.
2. Put the invariant evaluation statements into `unchecked` blocks, and added an underflow check where necessary.